app/services/sendgrid_mail_vo.rb

```diff
@@ -51,19 +53,67 @@ def get_vo
       subject: @subject,
       to: @to,
       to_name: @to_name,
-      dynamic_template_data: @dynamic_template_data
+      dynamic_template_data: @dynamic_template_data,
+      email_options: @email_options
     }
+    # # Add options, e.g., cc, bcc, reply_to, send_at, batch_id
+    # @email_options.each do |k, v|
+    #   mail_vo[k] = v
+    # end
+
     # Bug in Sendgrid's add_custom_arg.  Following will throw error "no implicit conversion of String into Hash"
     # mail.add_custom_arg(Hash["custom_arg" => {message_id: @message_id}])
     { mail_vo: mail_vo, message_id: @message_id }
   end
 
+  def self.add_options(mail_vo, mail, personalization)
+    email_options = JSON.parse(mail_vo[:email_options].to_json) # Convert from ActionController::Parameters
+    email_options.each do |k, v|
+      case k
+      when 'cc'
+        # ["recipient1@example.com <Alice Recipient>","recipient2@example.com"]
+        v.each do |i|
+          cc_maybe_name_ary = i.strip.gsub('  ', ' ').split(' ')
+          cc = cc_maybe_name_ary[0]
+          cc_name = cc_maybe_name_ary[1..-1].join(' ')[1..-2] if cc_maybe_name_ary.size > 1
+          personalization.add_cc(Email.new(email: cc, name: cc_name))
+        end
+      when 'bcc'
+        # ["recipient3@example.com","recipient4@example.com <Bob Recipient>"]
+        v.each do |i|
+          bcc_maybe_name_ary = i.strip.gsub('  ', ' ').split(' ')
```

Comment:
This is being used on`cc`, `bcc` and `reply_to`. We could have a function that return the name and the email.

For instance:

```ruby
def email_and_name_from_params(v)
    reply_to_maybe_name_ary = v.strip.gsub('  ', ' ').split(' ')
    reply_to = reply_to_maybe_name_ary[0]
    reply_to_name = reply_to_maybe_name_ary[1..-1].join(' ')[1..-2] if reply_to_maybe_name_ary.size > 1
    name, reply_to_name
end
```

And this function could be reused in the three areas.

```ruby
...
when 'cc'
    cc, cc_name = email_and_name_from_params(value)
    personalization.add_cc(Email.new(email: cc, name: cc_name))
when 'bcc'
...
```


With this, we could reuse it and not have to repeat it!